### ISSUE TITLE
Integer as name produces error

### ISSUE THREAD

**Issue by ajescode** [statsmodels/statsmodels#6826]
#### Describe the bug

Bug doesn't allow to perform HP filter on my specific data which name is integer instead of string.

Error:

```
File "/home/bonifacy/anaconda3/lib/python3.7/site-packages/statsmodels/tools/validation/validation.py", line 207, in wrap
    name = append if name is None else name + '_' + append
TypeError: unsupported operand type(s) for +: 'int' and 'str'
```

#### Code Sample, a copy-pastable example if possible

```python
data = hp.hpfilter(data, 6.25)
```

And my data:
```
0       0.107632
1      -1.097841
2      -0.523807
3       0.459228
4       1.040438
          ...   
1589    0.351596
1590   -0.409000
1591   -0.674491
1592    0.975859
1593    1.241350
Name: 7, Length: 1594, dtype: float64
```
Name is integer: 7

It works when we replace

`                name = append if name is None else name + '_' + append`
with
`                name = append if name is None else str(name) + '_' + append`

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

INSTALLED VERSIONS
------------------
Python: 3.7.6.final.0
OS: Linux 5.3.0-59-generic #53~18.04.1-Ubuntu SMP Thu Jun 4 14:58:26 UTC 2020 x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

statsmodels
===========

Installed: 0.11.0 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/statsmodels)

Required Dependencies
=====================

cython: 0.29.15 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/Cython)
numpy: 1.18.1 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/numpy)
scipy: 1.4.1 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/scipy)
pandas: 1.0.1 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/pandas)
    dateutil: 2.8.1 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/dateutil)
patsy: 0.5.1 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/patsy)

Optional Dependencies
=====================

matplotlib: 3.1.3 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/matplotlib)
    backend: Qt5Agg 
cvxopt: Not installed
joblib: 0.14.1 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/joblib)

Developer Tools
================

IPython: 7.12.0 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/IPython)
    jinja2: 2.11.1 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/jinja2)
sphinx: 2.4.0 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/sphinx)
    pygments: 2.5.2 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/pygments)
pytest: 5.3.5 (/home/bonifacy/anaconda3/lib/python3.7/site-packages/pytest)
virtualenv: Not installed

</details>

**Comment by bashtage** [comment]
Needs to call `str` before name to ensure that it is a string, or check if
a string and if not str, then treat as if missing.

**Comment by bashtage** [comment]
closed by f196c0963f1e9c85fea1e6eec2a00841cdce8a65
